### PR TITLE
Implement remind feature with user input criteria

### LINKED ISSUE #95: Implement remind feature to allow user input criteria
As an experienced-user, I want to have reminders for persons with upcoming premium due/policy expiry dates so that I can notify my clients about the upcoming payment.

The `remind` command allows users to see the clients that have expiry dates within their input days compared to the current date.
Example: `remind 30` will show the clients whose expiry date is within 30 days from the current date, `remind 60` will show the clients whose expiry date is within 60 days from the current date.

### PR DESCRIPTION
**Pull request by brendanneojw**
Closes #95 

Revamped remind command to now allow users to enter the number of days from the current date they want to see whose persons' policy expiry date is within that range.

Example:
`remind 30` shows persons with policy expiry dates that are within 30 days from the current date

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/RemindCommandParser.java [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
+                    String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, RemindCommand.MESSAGE_USAGE));
+        }
+
+        try {
+            int parsedArgs = Integer.parseInt(trimmedArgs);
+            if (parsedArgs < 0) {
+                throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
+                        Messages.MESSAGE_NOT_POSITIVE_NUMBER));
+            }
+            return new RemindCommand(new RemindPredicate(parsedArgs));
```
Comment (suryanshkushwaha) [inline src/main/java/seedu/address/logic/parser/RemindCommandParser.java:32]:
```suggestion
            
            return new RemindCommand(new RemindPredicate(parsedArgs));
```

File: src/main/java/seedu/address/logic/parser/RemindCommandParser.java [hunk 2]
```diff
@@ -1,17 +1,36 @@
...
+        String trimmedArgs = args.trim();
+
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, RemindCommand.MESSAGE_USAGE));
+        }
+
+        try {
+            int parsedArgs = Integer.parseInt(trimmedArgs);
+            if (parsedArgs < 0) {
```
Comment (suryanshkushwaha) [inline src/main/java/seedu/address/logic/parser/RemindCommandParser.java:27]:
```suggestion

            if (parsedArgs < 0) {
```

**Review by suryanshkushwaha — APPROVED** [review]

**Comment by suryanshkushwaha** [comment]
LGTM.

File: src/main/java/seedu/address/logic/commands/RemindCommand.java [hunk 3]
```diff
@@ -27,6 +33,7 @@ public RemindCommand(RemindPredicate remindPredicate) {
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
+        assert remindPredicate != null : "Predicate should never be null";
```
Comment (KumChaiYin) [inline src/main/java/seedu/address/logic/commands/RemindCommand.java:36]: Good in using assertion to check.

**Review by KumChaiYin — APPROVED** [review]
LGTM

**Pull request merged**